legion/engine/rendering/data/particlepolicies/flipbookpolicy.cpp

```diff
@@ -0,0 +1,39 @@
+#include <rendering/rendering.hpp>
+#include <rendering/data/particlepolicies/flipbookpolicy.hpp>
+
+namespace legion::rendering
+{
+    void flipbook_policy::setup(particle_emitter& emitter)
+    {
+        if (!emitter.has_buffer<scale>("scaleBuffer"))
+            emitter.create_buffer<scale>("scaleBuffer");
+        if (!emitter.has_buffer<size_type>("frameID"))
+            emitter.create_buffer<size_type>("frameID");
+
+        if (!emitter.has_uniform<int>("frameCount"))
+            emitter.create_uniform<int>("frameCount");
+    }
+
+    void flipbook_policy::onInit(particle_emitter& emitter, size_type start, size_type end)
+    {
+    }
+
+    void flipbook_policy::onUpdate(particle_emitter& emitter, float deltaTime, size_type count)
+    {
+        auto& frameIDBuffer = emitter.get_buffer<size_type>("frameID");
+        auto& ageBuffer = emitter.get_buffer<life_time>("lifetimeBuffer");
+        auto& frameCount = emitter.get_uniform<int>("frameCount");
+
+        for (size_type idx = 0; idx < count; idx++)
+        {
+            auto& lifeTime = ageBuffer[idx];
+            frameIDBuffer[idx] = (lifeTime.age / lifeTime.max) * frameCount;
```

Comment:
minor chance that `frameIDBuffer[idx]` is `frameCount` when in a very rare case `lifeTime.age == lifeTime.max`